2024-04-20 03:05:10,124 ERROR: Exception on /index [GET] [in C:\Users\dhana\anaconda3\envs\ML_Environment\lib\site-packages\flask\app.py:838]
Traceback (most recent call last):
  File "C:\Users\dhana\anaconda3\envs\ML_Environment\lib\site-packages\flask\app.py", line 1473, in wsgi_app
    response = self.full_dispatch_request()
  File "C:\Users\dhana\anaconda3\envs\ML_Environment\lib\site-packages\flask\app.py", line 882, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "C:\Users\dhana\anaconda3\envs\ML_Environment\lib\site-packages\flask\app.py", line 880, in full_dispatch_request
    rv = self.dispatch_request()
  File "C:\Users\dhana\anaconda3\envs\ML_Environment\lib\site-packages\flask\app.py", line 865, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)  # type: ignore[no-any-return]
  File "C:\Users\dhana\anaconda3\envs\ML_Environment\lib\site-packages\flask_login\utils.py", line 290, in decorated_view
    return current_app.ensure_sync(func)(*args, **kwargs)
  File "C:\Users\dhana\Documents\Bluechip Tecnologies Asia\The Flask Mega-Tutorial\Chapter-10\app\routes.py", line 45, in index
    return render_template('index.html', title='Home', form=form,
  File "C:\Users\dhana\anaconda3\envs\ML_Environment\lib\site-packages\flask\templating.py", line 149, in render_template
    template = app.jinja_env.get_or_select_template(template_name_or_list)
  File "C:\Users\dhana\anaconda3\envs\ML_Environment\lib\site-packages\jinja2\environment.py", line 1081, in get_or_select_template
    return self.get_template(template_name_or_list, parent, globals)
  File "C:\Users\dhana\anaconda3\envs\ML_Environment\lib\site-packages\jinja2\environment.py", line 1010, in get_template
    return self._load_template(name, globals)
  File "C:\Users\dhana\anaconda3\envs\ML_Environment\lib\site-packages\jinja2\environment.py", line 969, in _load_template
    template = self.loader.load(self, name, self.make_globals(globals))
  File "C:\Users\dhana\anaconda3\envs\ML_Environment\lib\site-packages\jinja2\loaders.py", line 137, in load
    code = environment.compile(source, name, filename)
  File "C:\Users\dhana\anaconda3\envs\ML_Environment\lib\site-packages\jinja2\environment.py", line 768, in compile
    self.handle_exception(source=source_hint)
  File "C:\Users\dhana\anaconda3\envs\ML_Environment\lib\site-packages\jinja2\environment.py", line 936, in handle_exception
    raise rewrite_traceback_stack(source=source)
  File "C:\Users\dhana\Documents\Bluechip Tecnologies Asia\The Flask Mega-Tutorial\Chapter-10\app\templates\index.html", line 26, in template
    {% endblock %}
jinja2.exceptions.TemplateSyntaxError: Encountered unknown tag 'endblock'. You probably made a nesting mistake. Jinja is expecting this tag, but currently looking for 'elif' or 'else' or 'endif'. The innermost block that needs to be closed is 'if'.
2024-04-20 03:05:12,203 ERROR: Exception on /explore [GET] [in C:\Users\dhana\anaconda3\envs\ML_Environment\lib\site-packages\flask\app.py:838]
Traceback (most recent call last):
  File "C:\Users\dhana\anaconda3\envs\ML_Environment\lib\site-packages\flask\app.py", line 1473, in wsgi_app
    response = self.full_dispatch_request()
  File "C:\Users\dhana\anaconda3\envs\ML_Environment\lib\site-packages\flask\app.py", line 882, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "C:\Users\dhana\anaconda3\envs\ML_Environment\lib\site-packages\flask\app.py", line 880, in full_dispatch_request
    rv = self.dispatch_request()
  File "C:\Users\dhana\anaconda3\envs\ML_Environment\lib\site-packages\flask\app.py", line 865, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)  # type: ignore[no-any-return]
  File "C:\Users\dhana\anaconda3\envs\ML_Environment\lib\site-packages\flask_login\utils.py", line 290, in decorated_view
    return current_app.ensure_sync(func)(*args, **kwargs)
  File "C:\Users\dhana\Documents\Bluechip Tecnologies Asia\The Flask Mega-Tutorial\Chapter-10\app\routes.py", line 187, in explore
    return render_template("index.html", title='Explore', posts=posts.items,
  File "C:\Users\dhana\anaconda3\envs\ML_Environment\lib\site-packages\flask\templating.py", line 149, in render_template
    template = app.jinja_env.get_or_select_template(template_name_or_list)
  File "C:\Users\dhana\anaconda3\envs\ML_Environment\lib\site-packages\jinja2\environment.py", line 1081, in get_or_select_template
    return self.get_template(template_name_or_list, parent, globals)
  File "C:\Users\dhana\anaconda3\envs\ML_Environment\lib\site-packages\jinja2\environment.py", line 1010, in get_template
    return self._load_template(name, globals)
  File "C:\Users\dhana\anaconda3\envs\ML_Environment\lib\site-packages\jinja2\environment.py", line 969, in _load_template
    template = self.loader.load(self, name, self.make_globals(globals))
  File "C:\Users\dhana\anaconda3\envs\ML_Environment\lib\site-packages\jinja2\loaders.py", line 137, in load
    code = environment.compile(source, name, filename)
  File "C:\Users\dhana\anaconda3\envs\ML_Environment\lib\site-packages\jinja2\environment.py", line 768, in compile
    self.handle_exception(source=source_hint)
  File "C:\Users\dhana\anaconda3\envs\ML_Environment\lib\site-packages\jinja2\environment.py", line 936, in handle_exception
    raise rewrite_traceback_stack(source=source)
  File "C:\Users\dhana\Documents\Bluechip Tecnologies Asia\The Flask Mega-Tutorial\Chapter-10\app\templates\index.html", line 26, in template
    {% endblock %}
jinja2.exceptions.TemplateSyntaxError: Encountered unknown tag 'endblock'. You probably made a nesting mistake. Jinja is expecting this tag, but currently looking for 'elif' or 'else' or 'endif'. The innermost block that needs to be closed is 'if'.
2024-04-22 14:47:59,047 INFO: Microblog startup [in C:\Users\dhana\Documents\Bluechip Tecnologies Asia\The Flask Mega-Tutorial\Chapter-10\app\__init__.py:46]
2024-04-22 14:49:49,146 ERROR: Exception on /index [GET] [in C:\Users\dhana\anaconda3\envs\ML_Environment\lib\site-packages\flask\app.py:838]
Traceback (most recent call last):
  File "C:\Users\dhana\anaconda3\envs\ML_Environment\lib\site-packages\flask\app.py", line 1473, in wsgi_app
    response = self.full_dispatch_request()
  File "C:\Users\dhana\anaconda3\envs\ML_Environment\lib\site-packages\flask\app.py", line 882, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "C:\Users\dhana\anaconda3\envs\ML_Environment\lib\site-packages\flask\app.py", line 880, in full_dispatch_request
    rv = self.dispatch_request()
  File "C:\Users\dhana\anaconda3\envs\ML_Environment\lib\site-packages\flask\app.py", line 865, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)  # type: ignore[no-any-return]
  File "C:\Users\dhana\anaconda3\envs\ML_Environment\lib\site-packages\flask_login\utils.py", line 290, in decorated_view
    return current_app.ensure_sync(func)(*args, **kwargs)
  File "C:\Users\dhana\Documents\Bluechip Tecnologies Asia\The Flask Mega-Tutorial\Chapter-10\app\routes.py", line 45, in index
    return render_template('index.html', title='Home', form=form,
  File "C:\Users\dhana\anaconda3\envs\ML_Environment\lib\site-packages\flask\templating.py", line 149, in render_template
    template = app.jinja_env.get_or_select_template(template_name_or_list)
  File "C:\Users\dhana\anaconda3\envs\ML_Environment\lib\site-packages\jinja2\environment.py", line 1081, in get_or_select_template
    return self.get_template(template_name_or_list, parent, globals)
  File "C:\Users\dhana\anaconda3\envs\ML_Environment\lib\site-packages\jinja2\environment.py", line 1010, in get_template
    return self._load_template(name, globals)
  File "C:\Users\dhana\anaconda3\envs\ML_Environment\lib\site-packages\jinja2\environment.py", line 969, in _load_template
    template = self.loader.load(self, name, self.make_globals(globals))
  File "C:\Users\dhana\anaconda3\envs\ML_Environment\lib\site-packages\jinja2\loaders.py", line 137, in load
    code = environment.compile(source, name, filename)
  File "C:\Users\dhana\anaconda3\envs\ML_Environment\lib\site-packages\jinja2\environment.py", line 768, in compile
    self.handle_exception(source=source_hint)
  File "C:\Users\dhana\anaconda3\envs\ML_Environment\lib\site-packages\jinja2\environment.py", line 936, in handle_exception
    raise rewrite_traceback_stack(source=source)
  File "C:\Users\dhana\Documents\Bluechip Tecnologies Asia\The Flask Mega-Tutorial\Chapter-10\app\templates\index.html", line 26, in template
    {% endblock %}
jinja2.exceptions.TemplateSyntaxError: Encountered unknown tag 'endblock'. You probably made a nesting mistake. Jinja is expecting this tag, but currently looking for 'elif' or 'else' or 'endif'. The innermost block that needs to be closed is 'if'.
